From 559c601625bb264b0ac0d069f1938496f7cb4d30 Mon Sep 17 00:00:00 2001
From: wschmidt <wschmidt@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 26 Jun 2014 18:50:25 +0000
Subject: [PATCH 070/134] 2014-06-26  Bill Schmidt 
 <wschmidt@linux.vnet.ibm.com>

	PR target/61542
	* config/rs6000/vsx.md (vsx_extract_v4sf): Fix bug with element
	extraction other than index 3.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@212050 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog            | 6 ++++++
 gcc/config/rs6000/vsx.md | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index 44896fb..784d3ca 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,9 @@
+2014-06-26  Bill Schmidt  <wschmidt@linux.vnet.ibm.com>
+
+	PR target/61542
+	* config/rs6000/vsx.md (vsx_extract_v4sf): Fix bug with element
+	extraction other than index 3.
+
 2014-06-24  Jakub Jelinek  <jakub@redhat.com>
 
 	PR target/61570
diff --git a/gcc/config/rs6000/vsx.md b/gcc/config/rs6000/vsx.md
index 792e687..d7839f5 100644
--- a/gcc/config/rs6000/vsx.md
+++ b/gcc/config/rs6000/vsx.md
@@ -1613,7 +1613,7 @@
     {
       if (GET_CODE (op3) == SCRATCH)
 	op3 = gen_reg_rtx (V4SFmode);
-      emit_insn (gen_vsx_xxsldwi_v4sf (op3, op1, op1, op2));
+      emit_insn (gen_vsx_xxsldwi_v4sf (op3, op1, op1, GEN_INT (ele)));
       tmp = op3;
     }
   emit_insn (gen_vsx_xscvspdp_scalar2 (op0, tmp));
-- 
1.8.5.5

